### PR TITLE
LibWeb: Make factory functions of our IDL implementations fallible

### PR DESCRIPTION
**Pull request by kennethmyhra**
This was split into small commits for my own sanity, and possibly the reviewer. If we want to squash them let me know

### CODE REVIEW

File: Userland/Libraries/LibWeb/SVG/SVGForeignObjectElement.cpp [hunk 1]
```diff
@@ -23,14 +24,24 @@ SVGForeignObjectElement::~SVGForeignObjectElement() = default;
...
+    }));
+    m_y = TRY(Web::Bindings::throw_dom_exception_if_needed(vm, [&]() {
+        return SVGAnimatedLength::create(realm, SVGLength::create(realm, 0, 0), SVGLength::create(realm, 0, 0));
+    }));
+    m_width = TRY(Web::Bindings::throw_dom_exception_if_needed(vm, [&]() {
+        return SVGAnimatedLength::create(realm, SVGLength::create(realm, 0, 0), SVGLength::create(realm, 0, 0));
+    }));
+    m_height = TRY(Web::Bindings::throw_dom_exception_if_needed(vm, [&]() {
+        return SVGAnimatedLength::create(realm, SVGLength::create(realm, 0, 0), SVGLength::create(realm, 0, 0));
+    }));
```
Comment (linusg) [inline Userland/Libraries/LibWeb/SVG/SVGForeignObjectElement.cpp:44]: omit `Web::`

File: Userland/Libraries/LibWeb/HTML/WorkerGlobalScope.cpp [hunk 2]
```diff
@@ -33,7 +34,9 @@ WorkerGlobalScope::~WorkerGlobalScope() = default;
 JS::ThrowCompletionOr<void> WorkerGlobalScope::initialize(JS::Realm& realm)
 {
     MUST_OR_THROW_OOM(Base::initialize(realm));
-    m_navigator = WorkerNavigator::create(*this);
+    m_navigator = TRY(Web::Bindings::throw_dom_exception_if_needed(realm.vm(), [&]() {
```
Comment (linusg) [inline Userland/Libraries/LibWeb/HTML/WorkerGlobalScope.cpp:37]: omit `Web::`

**Review by linusg — CHANGES_REQUESTED** [review]
Nice work! Not really happy with keeping the explicit dereferencing of NNGCPtr, which has an operator for that and is no longer needed after converting away from raw pointers.

I pointed out a handful of instances, but this applies to virtually all 70 commits (most commonly `something(*Foo::create())`) - please try your best 😅

For the next time such a repetitive task comes up I recommend starting with a handful, getting those reviewed, and then doing the rest with the previous feedback already included :)

**Comment by kennethmyhra** [comment]
Thanks! I will do my best to address it over the full patch set 🙇🏻‍♂️ 

And yeah the formatting issue, CLion really wanted to have it this way. I'll check out if clang-format accepts it differently.

**Review by kennethmyhra — COMMENTED** [review]
That should hopefully take care of all the dereferencing of NNGCPtr.

Squashed commit UIEvents::MouseEvent into DOM::Event since there were dependencies between them.

**Review by linusg — APPROVED** [review]
_nice_

**Pull request merged**